src/main/scala/uk/pkerrigan/darwin/Message.scala

```diff
@@ -0,0 +1,63 @@
+package uk.pkerrigan.darwin
+
+import scala.xml.{Node, XML}
+
+sealed trait Time
+case class EstimatedTime(time: String) extends Time
+case class ActualTime(time: String) extends Time
+
+sealed trait Location {
+  def tiploc: String
+}
+case class Station(tiploc: String, arrivalTime: Time, departureTime: Time) extends Location
+case class Waypoint(tiploc: String, passTime: Time) extends Location
+
+sealed trait Message {
+  def origin: String
+}
+case class ScheduleUpdate(origin: String, runId: String, locations: List[Location]) extends Message
+
+object Message {
+  def parseMessage(message: String): Option[Message] =
+    (XML.loadString(message) \ "uR")
+      .headOption
+      .flatMap(this.parseUpdateResponse)
+
+  private def parseUpdateResponse(updateResponse: Node): Option[Message] =
+    updateResponse.child
+      .headOption
+      .flatMap(this.parseMessageType(updateResponse.attribute("updateOrigin").head.text)(_))
```

Comment:
unsafe `.head` going on here, could explode

probably a for comprehension along the lines of 

```scala
for {
  child <- updateResponse.child.headOption
  origin <- child.attribute("updateOrigin").headOption
  result <- this.parseMessageType(origin.text)
} yield result
```